File: NetLibraryGenerator/NetLibraryGenerator/Model/LocalEntityDeclaration.cs
using NetLibraryGenerator.SchemeModel;
using NetLibraryGenerator.Utilities;

namespace NetLibraryGenerator.Model
{
    public class LocalEntityDeclaration
    {
        public int    Id          { get; set; }
        public string Name        { get; set; }
        public string LocalPath   { get; set; }
        public string Namespace   { get; set; }
        public ApiEntityKind Kind { get; set; }

        public LocalEntityDeclaration(ApiEntityDeclaration declaration)
        {
            Id = declaration.Id;
            Name = declaration.Name;
            Kind = declaration.Kind;

            string? parentFolder = Path.GetDirectoryName(declaration.Path);
            if (parentFolder == null) {
                LocalPath = Path.ChangeExtension(declaration.Path, "cs");
                Namespace = $"{Config.PROJECT_NAME}.{Config.MODEL_FOLDER_NAME}";
                return;
            }

            LocalPath = "";
            foreach (string node in parentFolder.Split('\\')) {
                LocalPath = Path.Combine(LocalPath, node.CaseTransform(Case.UnderScore, Case.PascalCase));
            }

            LocalPath = Path.Combine(LocalPath, $"{Path.GetFileNameWithoutExtension(declaration.Path)}.cs");
            Namespace = $"{Config.PROJECT_NAME}.{Config.MODEL_FOLDER_NAME}.{Path.GetDirectoryName(LocalPath)!.Replace('\\', '.')}";
        }
    }
}

File: DartLibraryGenerator/model/local_entity_declaration.dart
import 'package:path/path.dart' as path;
import 'package:recase/recase.dart';
import '../scheme_model/api_entity_declaration.dart';

class LocalEntityDeclaration {
  int id;       
  String name;     
  ApiEntityKind kind;
  late String localPath;

  LocalEntityDeclaration(ApiEntityDeclaration declaration) 
    : id = declaration.id, 
      name = declaration.name, 
      kind = declaration.kind,
      localPath = _toLocalPath(declaration.path);

  static String _toLocalPath(String entityPath) {
    String result = "";

    List<String> splittedPath = entityPath.split(r'\');
    for (var i = 0; i < splittedPath.length; i++) {
      if (i != splittedPath.length - 1) {
        result = path.join(result, splittedPath[i].snakeCase);
      } else {
        result = path.join(result, "${path.withoutExtension(splittedPath[i]).snakeCase}.dart");
      }
    }

    return result;
  }
}